src/main/java/Duke.java

```diff
@@ -1,10 +1,588 @@
+import java.io.*;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+import CustomException.DukeException;
+
+// These are not Strings; they are constant variables
+/* enum Op {
+    TODO,
+    DEADLINE,
+    EVENT,
+    LIST,
+    DONE,
+    DELETE,
+    BYE
+} */
+
+/**
+ * Represents a chatbot to store list of things to do. A <code>Duke</code> object corresponds to
+ * a chatbot that stores a list of things to do.
+ */
 public class Duke {
-    public static void main(String[] args) {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException | IOException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public static void main(String[] args) throws IOException {
+        Duke duke = new Duke("./src/main/java/database.txt");
+        duke.run();
+    }
+
+    public void run() throws IOException {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        ui.prompt(storage, tasks);
+    }
+}
+
+/**
+ * Represents a task to be done. It stores the description of the task and whether it has been
+ * done or not.
+ */
+class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Returns a symbol representing whether the task has been completed
+     *
+     * @return a tick (if done) or a cross (if not done)
+     */
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
+    }
+
+    /**
+     * Marks the task as done
+     */
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    /**
+     * Returns a string representation of the task
+     *
+     * @return a string representation of the task in the format "[X] Description"
+     */
+    @Override
+    public String toString() {
+        return "["
+                + getStatusIcon()
+                + "] "
+                + description;
+    }
+
+    /**
+     * Returns a string representation of the in the database format
+     *
+     * @return a string in the form "| 1 | description"
+     */
+    public String convert() {
+        int done = isDone ? 1 : 0;
+        return " | "
+                + done
+                + " | "
+                + description;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+}
```

Comment:
Shouldn't your classes such as Task, Event be split into separate files? Especially after the A-MoreOOP implementation and that you did it for Todo. Unless it's consistently inside the same Duke.java file i guess. Not too sure about that